src/convertToObject.js

```diff
@@ -3,16 +3,44 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
+
 function convertToObject(sourceString) {
-  // write your code here
+  const stylesArray = sourceString.split('\n');
+  const trimmed = [];
+
+  stylesArray.forEach((style) => {
+    trimmed.push(style.trim().replace(';', ''));
+  });
+
+  const cleared = [];
+
+  trimmed.forEach((e) => {
+    if (e) {
+      cleared.push(e.split(':'));
+    }
+  });
+
+  const final = cleared.reduce((res, [key, value]) => {
```

Comment:
In case value is undefined due to missing ':' in the input string, the function value.trim() will throw an error. You should add a check to ensure 'value' is not undefined before calling trim() on it.